src/chainstate/burn/mod.rs

```diff
@@ -188,6 +188,54 @@ impl SortitionHash {
     }
 }
 
+impl Opcodes {
+    const HTTP_BLOCK_COMMIT: &'static str = "block_commit";
+    const HTTP_KEY_REGISTER: &'static str = "key_register";
+    const HTTP_BURN_SUPPORT: &'static str = "burn_support";
+    const HTTP_STACK_STX: &'static str = "stack_stx";
+    const HTTP_PRE_STX: &'static str = "pre_stx";
+    const HTTP_TRANSFER_STX: &'static str = "transfer_stx";
+    const HTTP_DELEGATE_STX: &'static str = "delegate_stx";
+    const HTTP_PEG_IN: &'static str = "peg_in";
+
+    pub fn to_http_str(&self) -> &'static str {
+        match self {
+            Opcodes::LeaderBlockCommit => Self::HTTP_BLOCK_COMMIT,
+            Opcodes::LeaderKeyRegister => Self::HTTP_KEY_REGISTER,
+            Opcodes::UserBurnSupport => Self::HTTP_BURN_SUPPORT,
+            Opcodes::StackStx => Self::HTTP_STACK_STX,
+            Opcodes::PreStx => Self::HTTP_PRE_STX,
+            Opcodes::TransferStx => Self::HTTP_TRANSFER_STX,
+            Opcodes::DelegateStx => Self::HTTP_DELEGATE_STX,
+            Opcodes::PegIn => Self::HTTP_PEG_IN,
+        }
+    }
+
+    pub fn from_http_str(input: &str) -> Option<Opcodes> {
+        let opcode = if input == Self::HTTP_PEG_IN {
+            Opcodes::PegIn
+        } else if input == Self::HTTP_BLOCK_COMMIT {
+            Opcodes::LeaderBlockCommit
+        } else if input == Self::HTTP_KEY_REGISTER {
+            Opcodes::LeaderKeyRegister
+        } else if input == Self::HTTP_BURN_SUPPORT {
+            Opcodes::UserBurnSupport
+        } else if input == Self::HTTP_STACK_STX {
+            Opcodes::StackStx
+        } else if input == Self::HTTP_PRE_STX {
+            Opcodes::PreStx
+        } else if input == Self::HTTP_TRANSFER_STX {
+            Opcodes::TransferStx
+        } else if input == Self::HTTP_DELEGATE_STX {
+            Opcodes::DelegateStx
+        } else {
+            return None;
+        };
+
+        Some(opcode)
```

Comment:
Nit: This would be more readable as a match statement
```suggestion
    pub fn from_http_str(input: &str) -> Option<Opcodes> {
        Some(match input {
            Self::HTTP_PEG_IN => Opcodes::PegIn,
            Self::HTTP_BLOCK_COMMIT => Opcodes::LeaderBlockCommit,
            Self::HTTP_KEY_REGISTER => Opcodes::LeaderKeyRegister,
            Self::HTTP_BURN_SUPPORT => Opcodes::UserBurnSupport,
            Self::HTTP_STACK_STX => Opcodes::StackStx,
            Self::HTTP_PRE_STX => Opcodes::PreStx,
            Self::HTTP_TRANSFER_STX => Opcodes::TransferStx,
            Self::HTTP_DELEGATE_STX => Opcodes::DelegateStx,
            _ => return None,
        })
```

Reply:
Definitely! Updated in 40c1208c8